chemprop/cli/fingerprint.py

```diff
@@ -16,10 +29,187 @@ class FingerprintSubcommand(Subcommand):
+    if model_path.suffix in [".ckpt", ".pt"]:
+        return [model_path]
+    elif model_path.is_dir():
+        return list(model_path.rglob("*.ckpt")) + list(model_path.rglob("*.pt"))
+
+
+def make_fingerprint_for_model(
+    args: Namespace, model_path: Path, multicomponent: bool, output_path: Path
+):
+    model, input_scalers, output_scaler = load_model(
+        model_path, multicomponent
+    )  # TODO: connect input_scalers and output_scaler to the model
+    model.eval()
+
+    bounded = any(
+        isinstance(model.criterion, LossFunctionRegistry[loss_function])
+        for loss_function in LossFunctionRegistry.keys()
+        if "bounded" in loss_function
+    )
+
+    format_kwargs = dict(
+        no_header_row=args.no_header_row,
+        smiles_cols=args.smiles_columns,
+        rxn_cols=args.reaction_columns,
+        target_cols=None,
+        ignore_cols=None,
+        weight_col=None,
+        bounded=bounded,
+    )
+    featurization_kwargs = dict(
+        features_generators=args.features_generators, keep_h=args.keep_h, add_h=args.add_h
+    )
+
+    test_data = build_data_from_files(
+        args.test_path,
+        **format_kwargs,
+        p_descriptors=args.descriptors_path,
+        p_atom_feats=args.atom_features_path,
+        p_bond_feats=args.bond_features_path,
+        p_atom_descs=args.atom_descriptors_path,
+        **featurization_kwargs,
+    )
+    logger.info(f"test size: {len(test_data[0])}")
+    test_dsets = [
+        make_dataset(d, args.rxn_mode, args.multi_hot_atom_featurizer_mode) for d in test_data
+    ]
+
+    if multicomponent:
+        test_dset = data.MulticomponentDataset(test_dsets)
+    else:
+        test_dset = test_dsets[0]
+
+    X_d_scaler = input_scalers.get("X_d", None) if input_scalers else None
+    V_f_scaler = input_scalers.get("V_f", None) if input_scalers else None
+    E_f_scaler = input_scalers.get("E_f", None) if input_scalers else None
+    V_d_scaler = input_scalers.get("V_d", None) if input_scalers else None
+
+    if X_d_scaler is not None:
+        test_dset.normalize_inputs("X_d", X_d_scaler)
+    if V_f_scaler is not None:
+        test_dset.normalize_inputs("V_f", V_f_scaler)
+    if E_f_scaler is not None:
+        test_dset.normalize_inputs("E_f", E_f_scaler)
+    if V_d_scaler is not None:
+        test_dset.normalize_inputs("V_d", V_d_scaler)
+
+    test_loader = data.MolGraphDataLoader(
+        test_dset, args.batch_size, args.num_workers, shuffle=False
+    )
+
+    logger.info(model)
+
+    match RepresentationType.get(args.repr_type):
+        case RepresentationType.FINGERPRINT:
+            predictor = model.fingerprint
+            fingerprint_length = model.message_passing.output_dim
+        case RepresentationType.ENCODING:
+            predictor = model.encoding
+            fingerprint_length = model.predictor.ffn[-2][-1].out_features
+        case _:
+            raise RuntimeError("unreachable code reached!")
```

Comment:
`predictor` -> `func` or `encoding_function`. We're not making "predictions" here.